structuralcodes/core/base.py

```diff
@@ -48,3 +51,114 @@ def name(self):
     def density(self):
         """Returns the density of the material in kg/m3"""
         return self._density
+
+
+class ConstitutiveLaw(abc.ABC):
+    """Abstract base class for constitutive laws."""
+
+    constitutive_law_counter: t.ClassVar[int] = 0
+
+    def __init__(self, name: t.Optional[str] = None) -> None:
+        self.id = self.constitutive_law_counter
+        self._name = name if name is not None else f"ConstitutiveLaw_{self.id}"
+        self._increase_global_counter()
+
+    @property
+    def name(self):
+        """Returns the name of the constitutive law"""
+        return self._name
+
+    @classmethod
+    def _increase_global_counter(cls):
+        cls.constitutive_law_counter += 1
+
+    @abc.abstractmethod
+    def get_stress(self, eps: float) -> float:
+        """Each constitutive law should provide a method to return the
+        stress given the strain level"""
+
+    @abc.abstractmethod
+    def get_tangent(self, eps: float) -> float:
+        """Each constitutive law should provide a method to return the
+        tangent at a given strain level"""
+
+    def get_secant(self, eps: float) -> float:
```

Comment:
Should `get_secant` call `get_tanget` for `eps = 0.0` to avoid a `ZeroDivisionError`?